File: FinalProject/app/src/main/java/com/example/finalproject/fragments/TutorialFragment.java
package com.example.finalproject.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.SwitchCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.Switch;

import com.example.finalproject.R;

import org.jetbrains.annotations.NotNull;

public class TutorialFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_tutorial, container, false);
    }

    @Override
    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Switch sw=getView().findViewById(R.id.show_tutorial);
        sw.setChecked(true);
        sw.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity.show_tutorial=sw.isChecked();
            }
        });
        Button close_tutorial=getView().findViewById(R.id.close_tutorial);
        close_tutorial.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                MainActivity.show_tutorial=sw.isChecked();
                FragmentManager fm=getParentFragmentManager();
                FragmentTransaction fr=fm.beginTransaction();
                fr.add(R.id.status, new StatusBarFragment());
                fr.add(R.id.menu, new MenuFragment());
                fr.add(R.id.map, new MapFragment());
                fr.remove(fm.findFragmentById(R.id.tutorial));
                fr.commit();
            }
        });
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/fragments/TutorialFragment.kt
package com.example.finalproject.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.example.finalproject.MainActivity
import com.example.finalproject.R

class TutorialFragment : Fragment(), OnClickListener {

    private lateinit var closeTutorial: Button

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_tutorial, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        closeTutorial = requireView().findViewById(R.id.close_tutorial_button)
        closeTutorial.setOnClickListener(this)
    }

    override fun onClick(p0: View?) {
        MainActivity.showTutorial = false
        val fm = parentFragmentManager
        val fr = fm.beginTransaction()
        fr.add(R.id.status, StatusBarFragment())
        fr.add(R.id.menu, MenuFragment())
        fr.add(R.id.map, MapFragment())
        fr.remove(fm.findFragmentById(R.id.tutorial)!!)
        fr.commit()
    }
}